bundles/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/handler/WirelessChannelsHandler.java

```diff
@@ -0,0 +1,114 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.satel.internal.handler;
+
+import static org.openhab.binding.satel.internal.SatelBindingConstants.*;
+
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.binding.builder.ChannelBuilder;
+import org.eclipse.smarthome.core.thing.binding.builder.ThingBuilder;
+import org.openhab.binding.satel.internal.types.StateType;
+import org.openhab.binding.satel.internal.types.TroubleState;
+
+/**
+ * The {@link WirelessChannelsHandler} is base thing handler class for things that can be wireless devices.
+ * It adds support for two additional channels:
+ * <ul>
+ * <li><i>device_lobatt</i> - low battery indication</li>
+ * <li><i>device_nocomm</i> - communication problems indication</li>
+ * </ul>
+ * adding them if the device is configured as a wireless device.
+ *
+ * @author Krzysztof Goworek - Initial contribution
+ */
+public abstract class WirelessChannelsHandler extends SatelStateThingHandler {
```

Comment:
Why is this named `WirelessChannelsHandler` and not `WirelessDeviceHandler`?

Reply:
This handler adds common channels related to wireless devices, but regardless the device is wireless, or not. Also wired devices derive from this class and in such case this channels are removed. You can define whether the device is wireless or not in the thing configuration.

This is done in this way because you cannot detect a device is wireless or wired during discovery process. So all are added as wired and you can change them as wireless later on.

Maybe this not the best name, but haven't found better.